### ISSUE TITLE
 'iron create' command not working properly on Ubuntu 14.04

### ISSUE THREAD

**Issue by gitxav** [iron-meteor/iron-cli#233]
When running iron create project_name I get the following error:

Creating project  app... \
Meteor app created
created subastas0/app/client
created subastas0/app/client/.gitkeep
created subastas0/app/client/collections
created subastas0/app/client/collections/.gitkeep
created subastas0/app/client/head.html
created subastas0/app/client/lib
created subastas0/app/client/lib/.gitkeep
created subastas0/app/client/stylesheets
created subastas0/app/client/stylesheets/.gitkeep
created subastas0/app/client/templates
created subastas0/app/client/templates/.gitkeep
created subastas0/app/client/templates/layouts
created subastas0/app/client/templates/layouts/master_layout
created subastas0/app/client/templates/layouts/master_layout/master_layout.html
created subastas0/app/client/templates/layouts/master_layout/master_layout.js
created subastas0/app/client/templates/shared
created subastas0/app/client/templates/shared/loading
created subastas0/app/client/templates/shared/loading/loading.html
created subastas0/app/client/templates/shared/not_found
created subastas0/app/client/templates/shared/not_found/not_found.html
created subastas0/app/lib
created subastas0/app/lib/.gitkeep
created subastas0/app/lib/collections
created subastas0/app/lib/collections/.gitkeep
created subastas0/app/lib/controllers
created subastas0/app/lib/controllers/.gitkeep
created subastas0/app/lib/routes.js
created subastas0/app/packages
created subastas0/app/packages/.gitkeep
created subastas0/app/private
created subastas0/app/private/.gitkeep
created subastas0/app/public
created subastas0/app/public/.gitkeep
created subastas0/app/server
created subastas0/app/server/.gitkeep
created subastas0/app/server/bootstrap.js
created subastas0/app/server/collections
created subastas0/app/server/collections/.gitkeep
created subastas0/app/server/lib
created subastas0/app/server/lib/.gitkeep
created subastas0/app/server/publish.js
created app/client/templates/home
created app/client/templates/home/home.html
Error creating file ... Error: EISDIR: illegal operation on a directory, open '/home/dedalus/Meteorjs'
/usr/local/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
                        throw(ex);
                        ^

Error: EISDIR: illegal operation on a directory, open '/home/dedalus/Meteorjs'
    at Error (native)
    at Object.fs.openSync (fs.js:584:18)
    at Object.fs.writeFileSync (fs.js:1224:33)
    at Generator.createFile (/usr/local/lib/node_modules/iron-meteor/lib/tools/file_system.js:34:8)
    at Generator.template (/usr/local/lib/node_modules/iron-meteor/lib/tools/file_system.js:368:18)
    at Generator.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/generators/template.js:32:8)
    at Generator.invoke (/usr/local/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at /usr/local/lib/node_modules/iron-meteor/lib/generator.js:90:34
    at Object.CurrentConfig.withValue (/usr/local/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Object.CurrentConfig.withConfigFile (/usr/local/lib/node_modules/iron-meteor/lib/config.js:96:17)

I'm working on Ubuntu 14.04 with Meteor 1.2.1 and Nodejs 5.5.0

I'll really appreciate any advice to fix the issue.

**Comment by chriskgregory** [comment]
Same issue on Windows 8.1

**Comment by eliehamouche** [comment]
Any luck?
I'm having the exact same issue as well on OSX, Node 4.4.2, Meteor 1.3.

**Comment by comefectiva** [comment]
I'm having the same issue on Ubuntu 14.04, Node 4.4.2, Meteor 1.3.1, Iron 1.5.2

`/usr/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
						throw(ex);
						^

Error: EISDIR: illegal operation on a directory, unlink 'radiocoomevatv/app/client'
    at Error (native)
    at Object.fs.unlinkSync (fs.js:932:18)
    at /usr/lib/node_modules/iron-meteor/lib/tools/meteor.js:35:10
    at Function._.each._.forEach (/usr/lib/node_modules/iron-meteor/node_modules/underscore/underscore.js:153:9)
    at Command.createEmptyMeteorProject (/usr/lib/node_modules/iron-meteor/lib/tools/meteor.js:32:7)
    at /usr/lib/node_modules/iron-meteor/lib/commands/init.js:63:10
    at Object.CurrentConfig.withValue (/usr/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Command.<anonymous> (/usr/lib/node_modules/iron-meteor/lib/commands/init.js:58:24)
    at Command.invoke (/usr/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at Command.<anonymous> (/usr/lib/node_modules/iron-meteor/lib/commands/create.js:15:37)
`

**Comment by comefectiva** [comment]
This is a temporal solution:

If you cd into the app folder and execute **iron init** iron-cli can finish the installation successfully, you need to do this after the crash error.

This shoudn't be the right way but is a temporal solution.

**Comment by supphavit90** [comment]
@comefectiva yes, your solution works for me!

**Comment by rof20004** [comment]
@comefectiva Worked to mee.

**Comment by jospablos** [comment]
@comefectiva works for me as well, thanks!

**Comment by l0oky** [comment]
I am getting similar issue.

```bash
l0oky@laptop:~/Projects/nodejs/src/imasoft.ba$ iron create web
created web
created web/.iron
created web/.iron/config.json
created web/bin
created web/bin/.gitkeep
created web/build
created web/build/.gitkeep
created web/build/README
created web/config
created web/config/.gitkeep
created web/config/development
created web/config/development/env.sh
created web/config/development/settings.json
created web/config/production
created web/config/production/env.sh
created web/config/production/settings.json
Creating project  app... \
/usr/local/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
						throw(ex);
						^

Error: EISDIR: illegal operation on a directory, unlink 'web/app/client'
    at Error (native)
    at Object.fs.unlinkSync (fs.js:980:18)
    at /usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:35:10
    at Function._.each._.forEach (/usr/local/lib/node_modules/iron-meteor/node_modules/underscore/underscore.js:153:9)
    at Command.createEmptyMeteorProject (/usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:32:7)
    at /usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:63:10
    at Object.CurrentConfig.withValue (/usr/local/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Command.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:58:24)
    at Command.invoke (/usr/local/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at Command.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/commands/create.js:15:37)
```

I even tried @comefectiva solution:

```bash
l0oky@laptop:~/Projects/nodejs/src/imasoft.ba/web$ iron init
created .iron
created .iron/config.json
created bin
created bin/.gitkeep
created build
created build/.gitkeep
created build/README
created config
created config/.gitkeep
created config/development
created config/development/env.sh
created config/development/settings.json
created config/production
created config/production/env.sh
created config/production/settings.json
Creating project  app... |
/usr/local/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
						throw(ex);
						^

Error: EISDIR: illegal operation on a directory, unlink '/home/l0oky/Projects/nodejs/src/imasoft.ba/web/app/client'
    at Error (native)
    at Object.fs.unlinkSync (fs.js:980:18)
    at /usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:35:10
    at Function._.each._.forEach (/usr/local/lib/node_modules/iron-meteor/node_modules/underscore/underscore.js:153:9)
    at Command.createEmptyMeteorProject (/usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:32:7)
    at /usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:63:10
    at Object.CurrentConfig.withValue (/usr/local/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Command.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:58:24)
    at Command.invoke (/usr/local/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at Command.run (/usr/local/lib/node_modules/iron-meteor/lib/command.js:91:21)
```

What should I do? iron-meteor was working great a few days ago...

**Comment by danielschroeder25** [comment]
If I use @comefectiva solution I am getting:

Error: Command failed: /bin/sh -c meteor create app
    at Object.Future.wait (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/node_modules/fibers/future.js:420:15)
    at Command.execSync (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/tools/exec_sync.js:22:17)
    at Command.createEmptyMeteorProject (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/tools/meteor.js:31:10)
    at /home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/commands/init.js:63:10
    at Object.CurrentConfig.withValue (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Command.<anonymous> (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/commands/init.js:58:24)
    at Command.invoke (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at Command.run (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/command.js:91:21)
    at Command.runSubCommand (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/command.js:154:11)
    at Command.<anonymous> (/home/oxid/.npm-packages/lib/node_modules/iron-meteor/lib/iron.js:71:17)
    - - - - -
You can't create a Meteor project inside another Meteor project.

    at ChildProcess.exithandler (child_process.js:213:12)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)

**Comment by FrDarryl** [comment]
Running on DigitalOcean droplet. FWIW, Linux 4.4.0-22-generic #39-Ubuntu (16.04) x86_64

Scary to get such an error on scaffold creation!

Anyhow, the work-around fails if you run 'iron init' from the dir whose name is 'app', i.e., '/{project_name}/app'

You need to run 'iron init' in the project's root dir after creating it, i.e., with 'iron create {project-name}'. You'll also be asked by 'iron init' to confirm overwriting some files created with 'init create'.

**193:frdarryl@ifordparish:~/apps/
#** iron create ifordparish
created ifordparish
created ifordparish/.iron
created ifordparish/.iron/config.json
created ifordparish/bin
created ifordparish/bin/.gitkeep
created ifordparish/build
created ifordparish/build/.gitkeep
created ifordparish/build/README
created ifordparish/config
created ifordparish/config/.gitkeep
created ifordparish/config/development
created ifordparish/config/development/env.sh
created ifordparish/config/development/settings.json
created ifordparish/config/production
created ifordparish/config/production/env.sh
created ifordparish/config/production/settings.json
Creating project  app... |
/usr/local/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
                                                throw(ex);
                                                ^

Error: EISDIR: illegal operation on a directory, unlink 'ifordparish/app/client'
    at Error (native)
    at Object.fs.unlinkSync (fs.js:932:18)
    at /usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:35:10
    at Function._.each._.forEach (/usr/local/lib/node_modules/iron-meteor/node_modules/underscore/underscore.js:153:9)
    at Command.createEmptyMeteorProject (/usr/local/lib/node_modules/iron-meteor/lib/tools/meteor.js:32:7)
    at /usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:63:10
    at Object.CurrentConfig.withValue (/usr/local/lib/node_modules/iron-meteor/lib/config.js:65:13)
    at Command.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/commands/init.js:58:24)
    at Command.invoke (/usr/local/lib/node_modules/iron-meteor/lib/command.js:111:26)
    at Command.<anonymous> (/usr/local/lib/node_modules/iron-meteor/lib/commands/create.js:15:37)
**194:frdarryl@ifordparish:~/apps/
#** cd ifordparish/
**195:frdarryl@ifordparish:~/apps/ifordparish/
#** iron init
/home/frdarryl/apps/ifordparish/.iron/config.json already exists. Do you want to overwrite it? [yn]: y
created .iron/config.json
/home/frdarryl/apps/ifordparish/bin/.gitkeep already exists. Do you want to overwrite it? [yn]: y
created bin/.gitkeep
/home/frdarryl/apps/ifordparish/build/.gitkeep already exists. Do you want to overwrite it? [yn]: y
created build/.gitkeep
/home/frdarryl/apps/ifordparish/build/README already exists. Do you want to overwrite it? [yn]: y
created build/README
/home/frdarryl/apps/ifordparish/config/.gitkeep already exists. Do you want to overwrite it? [yn]: y
created config/.gitkeep
/home/frdarryl/apps/ifordparish/config/development/env.sh already exists. Do you want to overwrite it? [yn]: y
created config/development/env.sh
/home/frdarryl/apps/ifordparish/config/development/settings.json already exists. Do you want to overwrite it? [yn]: y
created config/development/settings.json
/home/frdarryl/apps/ifordparish/config/production/env.sh already exists. Do you want to overwrite it? [yn]: y
created config/production/env.sh
/home/frdarryl/apps/ifordparish/config/production/settings.json already exists. Do you want to overwrite it? [yn]: y
created config/production/settings.json
Meteor project already exists at "/home/frdarryl/apps/ifordparish/app"
created app/client/.gitkeep
created app/client/collections
created app/client/collections/.gitkeep
created app/client/head.html
created app/client/lib
created app/client/lib/.gitkeep
created app/client/stylesheets
created app/client/stylesheets/.gitkeep
created app/client/templates
created app/client/templates/.gitkeep
created app/client/templates/layouts
created app/client/templates/layouts/master_layout
created app/client/templates/layouts/master_layout/master_layout.html
created app/client/templates/layouts/master_layout/master_layout.js
created app/client/templates/shared
created app/client/templates/shared/loading
created app/client/templates/shared/loading/loading.html
created app/client/templates/shared/not_found
created app/client/templates/shared/not_found/not_found.html
created app/lib
created app/lib/.gitkeep
created app/lib/collections
created app/lib/collections/.gitkeep
created app/lib/controllers
created app/lib/controllers/.gitkeep
created app/lib/routes.js
created app/packages
created app/packages/.gitkeep
created app/private
created app/private/.gitkeep
created app/public
created app/public/.gitkeep
created app/server/.gitkeep
created app/server/bootstrap.js
created app/server/collections
created app/server/collections/.gitkeep
created app/server/lib
created app/server/lib/.gitkeep
created app/server/publish.js
created app/client/templates/home
created app/client/templates/home/home.html
created app/client/templates/home/home.js
created app/client/templates/home/home.css
created app/lib/controllers/home_controller.js
updated /home/frdarryl/apps/ifordparish/app/lib/routes.js
created app/lib/methods.js
created app/server/methods.js
Installing the package  iron:router... /
Changes to your project's package version selections:

iron:controller        added, version 1.0.12
iron:core              added, version 1.0.11
iron:dynamic-template  added, version 1.0.12
iron:layout            added, version 1.0.12
iron:location          added, version 1.0.11
iron:middleware-stack  added, version 1.0.11
iron:router            added, version 1.0.12
iron:url               added, version 1.0.11
reactive-dict          added, version 1.1.7


iron:router: Routing specifically designed for Meteor

✔ iron:router
created app/client/stylesheets/main.css
196:frdarryl@ifordparish:~/apps/ifordparish/

**Comment by vitaliy-kuzmich** [comment]
I have the same case. /usr/lib/node_modules/iron-meteor/node_modules/fibers/future.js:267
						throw(ex);
						^

Error: EISDIR: illegal operation on a directory, unlink 'postme-cli/app/client'

**Comment by badmark** [comment]
Followed instruction, after over writing files, I receive the following error:

```
Installing the package  iron:router... /
/home/mark/.nvm/versions/node/v6.5.0/lib/node_modules/iron-meteor/node_modules/fibers/future.js:300
						throw(ex);
Error: Command failed: meteor add iron:router
Problem! This project does not have a .meteor/release file. The file should
either contain the release of Meteor that you want to use, or the word 'none'
if you will only use the project with unreleased checkouts of Meteor. Please
edit the .meteor/release file in the project and change it to a valid Meteor
release or 'none'.
```

CD'ing into the 'app/' folder and running **meteor add iron:routing** returns the same '.meteor/release' error as above.

**Comment by chrisbutler** [comment]
this should be fixed now in 1.6.0
